contracts/LiquidityMiningCampaign.sol

```diff
@@ -0,0 +1,210 @@
+		_stakeAndLock(msg.sender,_tokenAmount, _lockScheme);
+
+	}
+
+	function _stakeAndLock(address _userAddress ,uint256 _tokenAmount, address _lockScheme) internal {
+		require(_tokenAmount > 0, "stakeAndLock::Cannot stake 0");
+
+		UserInfo storage user = userInfo[_userAddress];
+
+		uint256 userRewards = 0;
+
+		updateRewardMultipliers();
+		updateUserAccruedReward(_userAddress);
+
+		if(user.tokensOwed.length > 0) {
+			userRewards = user.tokensOwed[0];
+		}
+	
+		
+		for (uint256 i = 0; i < lockSchemes.length; i++) {
+
+			uint256 additionalRewards = calculateProportionalRewards(_userAddress, userRewards.sub(userAccruedRewads[_userAddress]), lockSchemes[i]);
+			LockScheme(lockSchemes[i]).updateUserAccruedRewards(_userAddress, additionalRewards);
+		}
+		userAccruedRewads[_userAddress]	= userRewards;
+		_stake(_tokenAmount, _userAddress, false);
+
+		lockToScheme(_lockScheme,_userAddress, _tokenAmount);
+
+		emit StakedAndLocked( _userAddress, _tokenAmount, _lockScheme);
+	}
+
+	function lockToScheme(address _lockScheme, address _userAddress, uint256  _tokenAmount) internal nonReentrant {
+		LockScheme(_lockScheme).lock(_userAddress, _tokenAmount );
+	}
+
+	function exitAndUnlock() public nonReentrant {
+			_exitAndUnlock(msg.sender);
+	}
+
+	function _exitAndUnlock(address _userAddress) internal {
+			UserInfo storage user = userInfo[_userAddress];
+			
+			
+			updateRewardMultipliers();
+			updateUserAccruedReward(_userAddress);
+			//todo check how to secure that 0 is the albt
+			uint256 finalRewards = user.tokensOwed[0].sub(userAccruedRewads[_userAddress]);
+		
+			for (uint256 i = 0; i < lockSchemes.length; i++) {
+
+				uint256 additionalRewards = calculateProportionalRewards(_userAddress, finalRewards, lockSchemes[i]);
+				if(additionalRewards > 0) {
+
+				LockScheme(lockSchemes[i]).updateUserAccruedRewards(_userAddress, additionalRewards);
+				}
+
+				uint256 bonus = LockScheme(lockSchemes[i]).exit(_userAddress);
+				IERC20Detailed(rewarsToken).safeTransfer(address(msg.sender), bonus);
+				
+			}
+			_exit(_userAddress);
+
+			// emit ExitedAndUnlocked(_userAddress);
+	}
+
+
+	function setReceiverWhitelisted(address receiver, bool whitelisted) override(StakeTransferer) onlyFactory public {
+		StakeTransferer.setReceiverWhitelisted(receiver, whitelisted);
+	}
+
+	/** @dev exits the current campaign and trasnfers the stake to another whitelisted campaign
+		@param transferTo address of the receiver to transfer the stake to
+	 */
+	function exitAndTransfer(address transferTo, address _lockScheme) virtual public onlyWhitelistedReceiver(transferTo) nonReentrant {
+		UserInfo storage user = userInfo[msg.sender];
+		
+		updateRewardMultipliers(); // Update the accumulated multipliers for everyone
+
+		if (user.amountStaked == 0) {
+			return;
+		}
+
+		updateUserAccruedReward(msg.sender); // Update the accrued reward for this specific user
+
+		uint256 bonus = LockScheme(_lockScheme).exit(msg.sender);
```

Comment:
As far as I understand `_lockScheme` is the scheme in the LMC that you are migrating to. In here you need a loop similar to the `exitAndUnlock`